File: python/difference-of-squares/difference_of_squares.py
def square_of_sum(number):
    numList = list(range(number+1)) 
    return sum(numList)**2

def sum_of_squares(number):
    numList = list(range(number+1))
    print("numList: ",numList)
    sum = 0
    for i in numList:
        sum += numList[i]**2
    print("sum:",sum)
    return sum

def difference_of_squares(number):
    x = square_of_sum(number)
    y = sum_of_squares(number)

    return x-y

File: cpp/difference-of-squares/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {

    int square_of_sum(int num){
        int sum = 0;
        for (int i = 0; i <= num; i++){
            sum += i;
        }
        return sum*sum;
    }
    
    int sum_of_squares(int num){
        int sum = 0;
        for (int i = 0; i <= num; i++){
            sum += i*i;
        }
        return sum;
    }

    int difference(int num){
        int square_sum = square_of_sum(num);
        int sum_square = sum_of_squares(num);
        return std::abs(square_sum - sum_square);
    }

}  // namespace difference_of_squares
